File: lib/typescript/botbuilder-skills/src/skillContext.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

/**
 * Context to share state between Bots and Skills.
 */
export class SkillContext {
    private readonly contextStorage: { [key: string]: Object };

    public constructor(contextStorage?: { [key: string]: Object }) {
        this.contextStorage = contextStorage || {};
    }

    public get count(): number {
        return Object.keys(this.contextStorage).length;
    }

    public getObj(key: string): Object|undefined {
        return this.contextStorage[key];
    }

    public setObj(key: string, value: Object): void {
        this.contextStorage[key] = value;
    }

    public forEachObj(func: (value: Object, key: string) => void): void {
        Object.entries(this.contextStorage)
            .forEach((v: [string, Object]): void => {
                func(v[1], v[0]);
            });
    }
}

File: sdk/csharp/libraries/microsoft.bot.builder.solutions/Skills/SkillContext.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using Newtonsoft.Json.Linq;

namespace Microsoft.Bot.Builder.Solutions.Skills
{
    /// <summary>
    ///  Context to share state between Bots and Skills.
    /// </summary>
    [Obsolete("This type is being deprecated. To continue using Skill capability please refer to https://aka.ms/botframework-solutions/releases/0_8", false)]
    public class SkillContext : Dictionary<string, JObject>
    {
        public SkillContext()
        {
        }

        public SkillContext(IDictionary<string, JObject> collection)
            : base(collection)
        {
        }
    }
}